atlas-core/src/main/java/io/atlasmap/core/AtlasMappingService.java

```diff
@@ -39,208 +43,303 @@
+
+        if (getJAXBContext() == null) {
+            setJAXBContext(JAXBContext.newInstance(stringListToColonSeparated(packages)));
+            if(logger.isDebugEnabled()) {
+                logger.debug("Initialized JAXBContext: " + stringListToColonSeparated(packages));
+            }
+        }
+
+        jaxbMarshaller = getJAXBContext().createMarshaller();
+        jaxbMarshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+        jaxbUnmarshaller = getJAXBContext().createUnmarshaller();
+        jsonMapper = new AtlasJsonMapper();
+    }
+    
+    public AtlasMapping loadMapping(String fileName) throws AtlasValidationException {
+        return loadMapping(fileName, AtlasMappingFormat.XML);
+    }
+    
+    public AtlasMapping loadMapping(String fileName, AtlasMappingFormat format) throws AtlasValidationException {
+        try {
+            AtlasMapping atlasMapping;
+            switch (format) {
+            case XML:
+                StreamSource streamSource = new StreamSource(new File(fileName));
+                atlasMapping = jaxbUnmarshaller.unmarshal(streamSource, AtlasMapping.class).getValue();
+                break;
+            case JSON:
+                atlasMapping = jsonMapper.readValue(new File(fileName), AtlasMapping.class);
+                break;
+            default:
+                throw new AtlasValidationException("Unsupported mapping format: " + format.value);
+            }
+            validate(atlasMapping);
+            return atlasMapping;
+        } catch (Exception e) {
+            throw new AtlasValidationException(e.getMessage(), e);
+        }
+    }
+    
+    public AtlasMapping loadMapping(Reader reader) throws AtlasValidationException {
+        return loadMapping(reader, AtlasMappingFormat.XML);
+    }
+    
+    public AtlasMapping loadMapping(Reader reader, AtlasMappingFormat format) throws AtlasValidationException {
+        try {
+            AtlasMapping atlasMapping;
+            switch (format) {
+            case XML:
+                StreamSource streamSource = new StreamSource(reader);
+                atlasMapping = jaxbUnmarshaller.unmarshal(streamSource, AtlasMapping.class).getValue();
+                break;
+            case JSON:
+                atlasMapping = jsonMapper.readValue(reader, AtlasMapping.class);
+                break;
+            default:
+                throw new AtlasValidationException("Unsupported mapping format: " + format.value);
+            }
+            validate(atlasMapping);
+            return atlasMapping;
+        } catch (Exception e) {
+            throw new AtlasValidationException(e.getMessage(), e);
+        }
+    }
+    
+    public AtlasMapping loadMapping(InputStream inputStream) throws AtlasValidationException {
+        return loadMapping(inputStream, AtlasMappingFormat.XML);
+    }
+    
+    public AtlasMapping loadMapping(InputStream inputStream, AtlasMappingFormat format) throws AtlasValidationException {
```

Comment:
See comment above. Sth like calling `loadMapping(new InputStreamReader(inputStream),format)` instead of doing the unmarshaling directly here.